### PR TITLE
add user auth functionnalities

### PR DESCRIPTION
**Pull request by raphaeldieu**
login / logout / signup with angular and basic front end to test it

installed dependencies : 
-    "angular-route": "~1.4.4",
-    "ng-token-auth": "~0.0.28",
-    "angular-cookies": "~1.4.4",
-    "bootstrap": "~3.3.5",
-    "jquery": "~2.1.4"

### CODE REVIEW

**Comment by raphaeldieu** [comment]
fixed merge conflicts

File: app/assets/javascripts/angular/application.js.erb [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
       templateUrl: "<%= asset_path('static_pages/index.html') %>",
       controller: 'StaticPagesCtrl'
+    }).
+    when('/signup', {
+      templateUrl: "<%= asset_path('auth/signup.html') %>",
+      controller: 'SignUpCtrl'
     })
+
+  $authProvider.configure({
+  apiUrl: 'http://localhost:3000'
```
Comment (h8rry) [inline app/assets/javascripts/angular/application.js.erb]: Don't forget to put the rails environment variable for your app to work on Heroku

File: app/assets/javascripts/angular/application.js.erb [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+  })
 }]);
```
Comment (h8rry) [inline app/assets/javascripts/angular/application.js.erb:18]: Don't forget to leave a blank line at the end of every file.

File: app/assets/javascripts/angular/controllers/static_pages.js.erb [hunk 3]
```diff
@@ -1,3 +1,31 @@
-app.controller('StaticPagesCtrl', ['$scope', function($scope){
-  $scope.message = "This comes from angular";
+app.controller('StaticPagesCtrl', ['$scope','$http','$auth','$location', function($scope,$http,$auth,$location){
```
Comment (h8rry) [inline app/assets/javascripts/angular/controllers/static_pages.js.erb]: `$scope, $http, $auth, $location` - spaces after each variable man!

**Comment by h8rry** [comment]
The rest looks fine

File: app/assets/javascripts/angular/application.js.erb [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
       templateUrl: "<%= asset_path('static_pages/index.html') %>",
       controller: 'StaticPagesCtrl'
+    }).
+    when('/signup', {
+      templateUrl: "<%= asset_path('auth/signup.html') %>",
+      controller: 'SignUpCtrl'
     })
+
+  $authProvider.configure({
+  apiUrl: <%= ENV['URL'] || 'http://localhost:3000' %>
```
Comment (raphaeldieu) [inline app/assets/javascripts/angular/application.js.erb:16]: added env variable

**Pull request merged**